docker/service/src/main/java/uk/nhs/adaptors/scr/utils/FhirHelper.java

```diff
@@ -17,4 +21,16 @@
             })
             .orElseThrow(() -> new FhirMappingException(resourceType.getSimpleName() + " missing from payload"));
     }
+
+    public static <T extends Resource> List<T> getDomainResourceList(Bundle bundle, Class<T> resourceType) {
```

Comment:
I think we already a similar class that Shuaib wrote.
First thing use java stream instead of that loop.
Second, don't use `Class<T>` but `ResourceType` enum instead. Each resource has `.getResourceType()` property